### PR TITLE
ci: add workflow that rejects merging jpeg and png images

### PR DESCRIPTION
**Pull request by f-hollow**
This PR adds a workflow to check if the changes in a PR have PNG and JPEG images. If yes, it blocks the PR for merging and asks to convert the images to WebP.

It also includes other smaller workflows:

- Adds a workflow that rejects filenames with spaces and asks, for example, to replace them with dashes.
- Adds instructions to the existing link checking workflow, because without them, it is not clear now what to do with links that failed automatic checking but are still valid (for example due to CAPTCHA or login requests).
- Adds a cron job to check links in all devportal content on the main branch every Monday.

### CODE REVIEW

**Review by pedrominatel — APPROVED** [review]

File: .github/workflows/cron-check-links.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Cron job to check links
+
+# Schedule to run the link checker every 5 minutes
```
Comment (horw) [inline .github/workflows/cron-check-links.yml:3]: Should be "At 02:00 on Monday.”

File: .github/workflows/pr-reject-png-jpg.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  reject-jpg-png:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Fetch all history to ensure we can access all branches
+
+      - name: Fetch main branch
```
Comment (horw) [inline .github/workflows/pr-reject-png-jpg.yml:16]: Just curious, is this step necessary? I see the previous step already fetches the data.

File: .github/workflows/cron-check-links.yml (hunk 1 above)
Comment (f-hollow) [reply .github/workflows/cron-check-links.yml:3]: Fixed!

File: .github/workflows/cron-check-links.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        uses: lycheeverse/lychee-action@v1
+        with:
+          args: |
+            --no-progress
+            --include-fragments
+            --exclude-path ./themes/
+            --exclude-path ./layouts/
+            .
+          # Fail action on broken links
+          fail: true
```
[the remark this answers is not in the document]
Comment (f-hollow) [reply .github/workflows/cron-check-links.yml:26]: Done

File: .github/workflows/pr-reject-png-jpg.yml (hunk 2 above)
Comment (f-hollow) [reply .github/workflows/pr-reject-png-jpg.yml:16]: You are completely right! Removed this step here and in `pr-check-file-names`.

**Review by horw — COMMENTED** [review]
LGTM

**Review by horw — APPROVED** [review]

**Pull request merged**